ui-v2/app/components/healthcheck-list.js

```diff
@@ -0,0 +1,36 @@
+import Component from '@ember/component';
+import { get } from '@ember/object';
+
+export default Component.extend({
+  // TODO: Could potentially do this on attr change
+  actions: {
+    sortChecksByImportance: function(a, b) {
+      const statusA = get(a, 'Status');
+      const statusB = get(b, 'Status');
+      switch (statusA) {
+        case 'passing':
+          // a = passing
+          // unless b is also passing then a is less important
+          return statusB === 'passing' ? 0 : 1;
+        case 'critical':
+          // a = critical
+          // unless b is also critical then a is more important
+          return statusB === 'critical' ? 0 : -1;
+        case 'warning':
+          // a = warning
+          switch (statusB) {
+            // b is passing so a is more important
+            case 'passing':
+              return -1;
+            // b is critical so a is less important
+            case 'critical':
+              return 1;
+            // a and b are both warning, therefore equal
+            default:
+              return 0;
+          }
+      }
+      return 0;
+    },
+  },
+});
```

Comment:
This component is now used to render healthcheck 'outputs' i.e. the log information and further detail for the healthchecks, it basically groups `{{healthcheck-output}}`. These 2 components have been created as we are now rendering this list of healthcheck outputs in various places in the UI. I've left the old implementation in the Nodes Detail https://github.com/hashicorp/consul/blob/fc1d9e5d78749edc55249e5e7c1a8f7a24add99d/ui-v2/app/controllers/dc/nodes/show.js#L56 page for the sake of not touching what we are doing in this PR. But this is one of the areas we'll be looking to refactor near term.